packages/at_client/test/shared_key_encryption_test.dart

```diff
@@ -0,0 +1,194 @@
+    var aesSharedKey = '7l00PvmXMD9i1z0Q72O7RNQc6D9/9k9FrqfvCZcEBqs=';
+    var sharedKeyEncryptedWithAlicePublicKey =
+        EncryptionUtil.encryptKey(aesSharedKey, aliceEncryptionPublicKey);
+    var sharedKeyEncryptedWithBobPublicKey =
+        EncryptionUtil.encryptKey(aesSharedKey, bobEncryptionPublicKey);
+    var bobPublicKeyCheckSum =
+        EncryptionUtil.md5CheckSum(bobEncryptionPublicKey);
+    var location = 'California';
+    var atEncryptionKeyPair = AtEncryptionKeyPair.create(
+        aliceEncryptionPublicKey, aliceEncryptionPrivateKey);
+
+    AtChopsKeys atChopsKeys = AtChopsKeys.create(atEncryptionKeyPair, null);
+    var atChopsImpl = AtChopsImpl(atChopsKeys);
+    when(() => mockAtClient.atChops).thenAnswer((_) => atChopsImpl);
+    when(() => mockAtClient.getCurrentAtSign()).thenReturn('@alice');
+    when(() => mockLocalSecondary.getEncryptionPublicKey('@alice'))
+        .thenAnswer((_) => Future.value(aliceEncryptionPublicKey));
+    when(() => mockLocalSecondary.executeVerb(any<LLookupVerbBuilder>()))
+        .thenAnswer((Invocation invocation) {
+      final builder = invocation.positionalArguments[0] as LLookupVerbBuilder;
+      final buildKeyValue = builder.buildKey();
+      if (buildKeyValue == 'shared_key.bob@alice') {
+        return Future.value('data:$sharedKeyEncryptedWithAlicePublicKey');
+      } else if (buildKeyValue == 'cached:public:publickey@bob') {
+        return Future.value('data:$bobEncryptionPublicKey');
+      } else if (buildKeyValue == '@bob:shared_key@alice') {
+        return Future.value('data:$sharedKeyEncryptedWithBobPublicKey');
+      } else {
+        return Future.value('data:null');
+      }
+    });
+    var sharedKey = AtKey()
+      ..sharedBy = '@alice'
+      ..sharedWith = '@bob'
+      ..key = 'location';
+    sharedKey.metadata = (Metadata()..pubKeyCS = bobPublicKeyCheckSum);
+    var encryptionResult =
+        await sharedKeyEncryption.encrypt(sharedKey, location);
+    expect(encryptionResult != location, true);
+    var decryptionResult =
+        await sharedKeyDecryption.decrypt(sharedKey, encryptionResult);
+    expect(decryptionResult, location);
+  });
+
+  test('test to check shared key encryption - with IV', () async {
```

Comment:
Please add a brief description of what this test is testing

Reply:
done